src/deflate/mod.rs

```diff
@@ -45,10 +42,7 @@ enum BlockType {
 mod tests {
     use super::*;
     use crate::lz77;
-    #[cfg(feature = "no_std")]
     use core2::io::{Read, Write};
```

Comment:
```suggestion
    use core2::io::{Read, Write};
    #[cfg(not(feature = "std"))]
    use alloc::vec::Vec;
```

Reply:
No, there's no reason to do this - you can always import from `alloc::vec::Vec` same as for items in `core`.

Reply:
> No, there's no reason to do this

There absolutely is a reason to do this.

With `no_std` test builds you need to explicitly import from `alloc`, with `std` builds the `Vec` struct is automatically imported from the `prelude`. If you want to remove the conditional compilation, that's fine. However, you absolutely need the `use alloc::vec::Vec` import, either way.

You don't have to take my word for it, try running the tests under `no_std` and `std` builds:

```
# no_std tests
cargo test --no-default-features --all

# std tests
cargo test --all
```

And notice all the build errors, which the comments in this PR address.

Reply:
This is what I get for haphazardly writing on my phone before a meeting and forgetting to follow up. 🤷🏻‍♀️ 

> No, there's no reason to do this

I was specifically referring to the conditional compilation, which is indeed unnecessary. You can always `use alloc::vec::Vec`, which is what I was getting at.

You're entirely correct about the build errors and needing to import `Vec` in the first place. Somehow I missed actually running `cargo test --no-default-features --all` the other night? Good catch, my apologies on both missing that and responding in a poor way 🙃

Reply:
No worries :smile:

Reply:
Done.